### DISCUSSION TITLE
Is wezterm.sh required?

### DISCUSSION THREAD

**Discussion by mawaldne** [wezterm/wezterm#3690] [General]
Hi, should we be sourcing [wezterm.sh](https://github.com/wez/wezterm/blob/main/assets/shell-integration/wezterm.sh) into our shell environments? I've seen it mentioned in some [other discussions](https://github.com/wez/wezterm/issues/115).

I see inconsistent behavior when trying to use the [get_current_working_dir](https://wezfurlong.org/wezterm/config/lua/pane/get_current_working_dir.html?h=get_current_working_dir) function on macOS. Sometime it seems to work, but other times it seems delayed (like it uses the cwd from a command ago). I think this is because of OSC 7 as mentioned in the docs.  When I source `wezterm.sh`, `get_current_working_dir` works every time. 

So, should `wezterm.sh` just be sourced automatically as part of the regular install? Or should there be some note as part of the install docs saying you should source this file.

Thank you!

**Comment by wez** [comment]
It's not required, but you may have a superior experience if you do.
On macOS, there isn't a great story for automatically installing shell integrations, which is one of the reasons why this isn't pushed very hard.
